### PR TITLE
Prim api gen

### PR DESCRIPTION
**Pull request by xiaoguoguo626807**
### PR types
Others 

### PR changes
Others

### Describe
算子组合机制中prim_api.h 及 eager_ptim_api.cc 自动代码生成文件，依赖于parse.yaml 解析。

### CODE REVIEW

File: paddle/fluid/prim/api/manual/backward/composite_backward_api.h [hunk 1]
```diff
@@ -13,7 +13,7 @@
 // limitations under the License.
 
 #pragma once
-#include "paddle/fluid/prim/api/manual/prim_api/prim_api.h"
```
Comment (JiabinYang) [inline paddle/fluid/prim/api/manual/backward/composite_backward_api.h:16]: Don't change this just add new header for generated api

File: paddle/fluid/prim/api/generated/prim_api/static_prim_api.cc [hunk 2]
```diff
@@ -25,7 +25,7 @@
 #include "paddle/fluid/framework/program_desc.h"
 
 #include "paddle/fluid/framework/convert_utils.h"
-#include "paddle/fluid/prim/api/manual/prim_api/prim_api.h"
+#include "paddle/fluid/prim/api/generated/prim_api/prim_api.h"
```
Comment (JiabinYang) [inline paddle/fluid/prim/api/generated/prim_api/static_prim_api.cc:28]: same

File: paddle/fluid/prim/api/generated/prim_api/CMakeLists.txt [hunk 3]
```diff
@@ -7,5 +7,5 @@ if(NOT (NOT WITH_PYTHON AND ON_INFER))
   cc_library(
     eager_prim_api
     SRCS eager_prim_api.cc
-    DEPS final_dygraph_function eager_utils)
+    DEPS eager_utils)
```
Comment (JiabinYang) [inline paddle/fluid/prim/api/generated/prim_api/CMakeLists.txt]: why remove this

**Review by JiabinYang — COMMENTED** [review]
some comments

File: paddle/fluid/prim/api/manual/backward/composite_backward_api.h (hunk 1 above)
Comment (xiaoguoguo626807) [reply paddle/fluid/prim/api/manual/backward/composite_backward_api.h:16]: done

File: paddle/fluid/prim/api/generated/prim_api/static_prim_api.cc (hunk 2 above)
Comment (xiaoguoguo626807) [reply paddle/fluid/prim/api/generated/prim_api/static_prim_api.cc:28]: done

File: paddle/fluid/prim/api/generated/prim_api/CMakeLists.txt (hunk 3 above)
Comment (xiaoguoguo626807) [reply paddle/fluid/prim/api/generated/prim_api/CMakeLists.txt]: recovered

File: paddle/fluid/prim/api/auto_code_generated/prim_base.py [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+        return self.api
+
+    # def is_inplace(self):
+    #     if self.inplace_map
+    #         return True
+    #     return False
+
+    def get_input_tensor_args(self, inplace_flag=False):
+        input_args = []
+        inplace_type_map = {
```
Comment (JiabinYang) [inline paddle/fluid/prim/api/auto_code_generated/prim_base.py:82]: Maybe don't need this map for now?

**Review by JiabinYang — APPROVED** [review]
LGTM

**Pull request merged**